vault/token_store.go

```diff
@@ -1076,6 +1076,11 @@ func (ts *TokenStore) handleCreateCommon(
 			logical.ErrInvalidRequest
 	}
 
+	// Prevent attempts to create a root token without an actual root token as parent
```

Comment:
I think the placement of this code is much earlier than where this is supposed to be.
`data.Policies` and hence `te.Policies` get modified further down as well.
This check should be placed at least after 1187, and the check can be performed directly on `te.Policies`